app/src/main/java/net/gsantner/markor/activity/MainActivity.java

```diff
@@ -104,45 +109,46 @@ public void onPageSelected(int position) {
 
         // Switch to tab if specific folder _not_ requested, and not recreating from saved instance
         final int startTab = _appSettings.getAppStartupTab();
-        if (startTab != R.id.nav_notebook && savedInstanceState == null && getIntentDir(getIntent(), null) == null) {
+        if (startTab != R.id.nav_notebook && savedInstanceState == null && GsFileUtils.getValidIntentDir(getIntent(), null) == null) {
             _viewPager.postDelayed(() -> _viewPager.setCurrentItem(tabIdToPos(startTab)), 100);
         }
     }
 
+    // Reduces swipe sensitivity
+    // Inspired by https://stackoverflow.com/a/72067439
+    private void reduceViewpagerSwipeSensitivity() {
+        final int SLOP_MULTIPLIER = 4;
+        try
+        {
+            final Field ff = ViewPager2.class.getDeclaredField("mRecyclerView");
+            ff.setAccessible(true);
+            final RecyclerView recyclerView = (RecyclerView) ff.get(_viewPager);
+            // Set a constant so we don't continuously reduce this value with every call
+            recyclerView.setScrollingTouchSlop(RecyclerView.TOUCH_SLOP_PAGING);
+            final Field touchSlopField = RecyclerView.class.getDeclaredField("mTouchSlop");
+            touchSlopField.setAccessible(true);
+            final int touchSlop = (int) touchSlopField.get(recyclerView);
+            touchSlopField.set(recyclerView, touchSlop * SLOP_MULTIPLIER);
+        } catch(NoSuchFieldException e) {
+            Log.d(MainActivity.class.getName(), e.getMessage());
+        } catch(IllegalAccessException e) {
+            Log.d(MainActivity.class.getName(), e.getMessage());
+        } catch (NullPointerException e) {
```

Comment:
Enough to have one catchall. Not wanted to crash something here for that hack. And whatever error happens, it means the workaround doesn't work anymore and all we can do is print a warning